cpp/src/arrow/array/concatenate_test.cc

```diff
@@ -661,14 +663,104 @@ TEST_F(ConcatenateTest, ExtensionType) {
   });
 }
 
+std::shared_ptr<DataType> LargeVersionOfType(const std::shared_ptr<DataType>& type) {
+  switch (type->id()) {
+    case Type::BINARY:
+      return large_binary();
+    case Type::STRING:
```

Comment:
So view type would not meet the problem here? The new append data will point to a new "buf. index" rather than concatenate togather in the previous buffer?

Reply:
The casts coverage for string-view is not perfect yet.

Reply:
https://github.com/apache/arrow/issues/42247